Index: flac-1.2.1/src/metaflac/options.c
===================================================================
--- flac-1.2.1.orig/src/metaflac/options.c
+++ flac-1.2.1/src/metaflac/options.c
@@ -1041,7 +1041,7 @@ FLAC__bool parse_block_type(const char *
 			out->entries[entry].filter_application_by_id = (0 != r);
 			if(0 != r) {
 				if(strlen(r) == 4) {
-					strcpy(out->entries[entry].application_id, r);
+					memcpy(out->entries[entry].application_id, r, 4);
 				}
 				else if(strlen(r) == 10 && strncmp(r, "0x", 2) == 0 && strspn(r+2, "0123456789ABCDEFabcdef") == 8) {
 					FLAC__uint32 x = strtoul(r+2, 0, 16);
